### DISCUSSION TITLE
Refactor of `registerComponentsFromDirectory` from `FilamentServiceProvider`

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#5901] [Ideas]
Possibly thinking aloud here. I've been using the method `registerComponentsFromDirectory` in a couple of plugins I've created recently. Basically, I've had to copy the code into my own service provider since it is in `FilamentServiceProvider`, `protected` and not in `PluginServiceProvider` from which I extend my own Service Providers.

https://github.com/filamentphp/filament/blob/1473a77f3148edb83ca063354dd1dcccb540ce3f/packages/admin/src/FilamentServiceProvider.php#L205

At its core, this method is essentially a helper method to identifiy any classes that implement a given class in a given location. This method is great and a huge time saver for me. It allows me to collate and register custom entities that I want to use in my own plugins/apps.

Would it it be feasible/desirable to move this method to `PluginServiceProvider` so that plugin developers can make use of it? Happy to PR. Not sure if this would be deemed as a breaking change. AFAICT it is only used in `FilamentServiceProvider itself so impact should be minial internally.

**Comment by danharrin** [comment]
It's not a breaking change, but this method is gone in v3 anyway so I'm not sure that it's worth it?

**Reply by titantwentyone** [reply]
Ah yeh. Can see that. It's been modified and put into `HasComponents` I believe as `discoverComponents`. No worries. Thanks!
